File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/smt/model/SList.java
package com.github.jmlparser.smt.model;

import com.github.javaparser.ast.type.Type;
import com.github.javaparser.resolution.types.ResolvedType;

import java.io.PrintWriter;
import java.util.Objects;

/**
 * @author Alexander Weigl
 * @version 1 (07.08.22)
 */
public class SList extends SExpr {
    private SExpr[] value;

    public SList(SmtType stype, ResolvedType javaType, SExpr... args) {
        this.smtType = stype;
        this.javaType = javaType;
        this.value = args;
        for (SExpr arg : args) {
            Objects.requireNonNull(arg);
        }
    }

    public SExpr[] getValue() {
        return value;
    }

    @Override
    public void appendTo(PrintWriter writer) {
        writer.write('(');
        for (int i = 0; i < value.length; i++) {
            value[i].appendTo(writer);
            if (i < value.length - 1)
                writer.write(' ');
        }
        writer.write(')');
    }

    public SExpr get(int i) {
        return value[i];
    }
}

File: tools/smt/src/main/kotlin/io/github/jmltoolkit/smt/model/SList.kt
package io.github.jmltoolkit.smt.model

import com.github.javaparser.resolution.types.ResolvedType
import java.io.PrintWriter

/**
 * @author Alexander Weigl
 * @version 1 (07.08.22)
 */
class SList(stype: SmtType?, javaType: ResolvedType?, private val value: List<SExpr>) : SExpr(javaType, stype) {
    override fun appendTo(writer: PrintWriter) {
        writer.write('('.code)
        for (i in value.indices) {
            value[i].appendTo(writer)
            if (i < value.size - 1) writer.write(' '.code)
        }
        writer.write(')'.code)
    }

    fun get(i: Int): SExpr {
        return value[i]
    }
}
